File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        st=[]
        for ch in s:
            if ch in '[{(':
                st.append(ch)
            else:
                if len(st)==0:
                    return False
                p=st.pop()
                if (ch=='}'and p!='{')or(ch==']'and p!='[')or(ch==')'and p!='('):
                    return False
        return len(st)==0

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(var ch : s.toCharArray()){
            if("[{(".contains(ch+""))
                st.push(ch);
            else{
                if(st.isEmpty())
                    return false;
                char p=st.pop();
                if((ch=='}'&& p!='{')||(ch==']'&& p!='[')||(ch==')'&& p!='('))
                    return false;
            }
        }

        return st.isEmpty();   
    }
}